client/src/pages/Feed.js

```diff
@@ -466,15 +564,44 @@ const Feed = (props) => {
       }
     };
 
+    const searchURL = () => {
+      if (searchKeyword)
+        return `&keywords=${encodeURIComponent(searchKeyword)}`;
+      else return "";
+    };
+
     const limit = PAGINATION_LIMIT;
     const skip = page * limit;
-    const baseURL = `/api/posts?&includeMeta=true&limit=${limit}&skip=${skip}`;
-    let endpoint = `${baseURL}${objectiveURL()}${filterURL()}`;
+    let baseURL = `/api/posts?includeMeta=true&limit=${limit}&skip=${skip}`;
+    switch (searchCategory) {
+      case "POSTS":
+        break;
+      case "INDIVIDUALS":
+        baseURL = `/api/users?includeMeta=true&limit=${limit}&skip=${skip}`;
+        break;
+      case "ORGANISATIONS":
+        baseURL = `/api/organisations/search?includeMeta=true&limit=${limit}&skip=${skip}`;
+        break;
+      default:
+        break;
+    }
+
+    let endpoint = `${baseURL}${objectiveURL()}${filterURL()}${searchURL()}`;
     postsDispatch({ type: FETCH_POSTS });
+
     try {
       const {
         data: { data: posts, meta },
       } = await axios.get(endpoint);
+      if (searchKeyword) {
+        // trigger custom GTM event
+        window.dataLayer.push({
```

Comment:
This breaks for me (did you only get it to work locally for yourself @Zaydme ?).

![image](https://user-images.githubusercontent.com/10546720/97351568-519f6c80-1868-11eb-80a9-5a81ff8a7dc7.png)


Do you need similar to `client/src/actions/authActions.js` 

```
        // trigger custom GTM event
        TagManager.dataLayer({
          dataLayer: {
            event: "SEARCH_KEYWORD",
            keyword: searchKeyword,
            category: searchCategory || "POSTS",
            resultsCount: meta.total,
          }
        });
```

This no longer breaks for me BUT I don't know if it works the way intended with GTM.

With imports:

```
import TagManager from "react-gtm-module";
```

Reply:
do you have `REACT_APP_GTM_ID`?

Reply:
@joshmorel oh you have QA mode enabled, so GTM is not initialized, I just committed a fix to make sure the dataLayer exists before pushing

Reply:
> do you have REACT_APP_GTM_ID?
> oh you have QA mode enabled, so GTM is not initialized

@Zaydme QA mode wasn't enabled but like most devs I'm not working with a `GTM_APP_GTM_ID` (both will cause this issue however as GTM is not initialized). 

> I just committed a fix to make sure the dataLayer exists before pushing

This will work but such a check must also be included when using `TagManager.dataLayer` (react library). Is there anything else it might do vs using window.dataLayer directly? @idjevm  Probably not much but we should be sure.

Reply:
@joshmorel
I think that's what the GTM library does, because I just followed the GTM documentations and what they do is pushing an object to `window.dataLayer` 
I tested with the staging GTM and everything seems to be working.